src/sentry/auth/helper.py

```diff
@@ -507,7 +507,17 @@ def handle_new_user(auth_provider, organization, request, identity):
 
     user.send_confirm_emails(is_new_user=True)
 
-    handle_new_membership(auth_provider, organization, request, auth_identity)
+    # If the user has a pending invitation in this organization, we can
+    # immediately assocaite their user.
+    try:
+        member = OrganizationMember.objects.get(
+            organization=organization,
+            email=user.email,
```

Comment:
Should the email by set to null after this?

Reply:
`member.set_user` (perhaps deceptively) does that for you actually